bitnami/keycloak/templates/statefulset.yaml

```diff
@@ -204,6 +204,8 @@ spec:
                   name: {{ $globalSecretName }}
                   key: {{ include "common.secrets.key" (dict "existingSecret" .Values.auth.existingSecret "key" "management-password") }}
                 {{- end }}
+            - name: DB_VENDOR
+              value: {{ .Values.externalDatabase.vendor }}
```

Comment:
I don't think this is enough
This env. variable (`DB_VENDOR`) isn't taken into consideration by the container init logic that will configure Keycloak to use PostgreSQL regardless this value. See:

- https://github.com/bitnami/bitnami-docker-keycloak/blob/master/15/debian-10/rootfs/opt/bitnami/scripts/libkeycloak.sh#L102
- https://github.com/bitnami/bitnami-docker-keycloak/blob/master/15/debian-10/rootfs/opt/bitnami/scripts/libkeycloak.sh#L108
- https://github.com/bitnami/bitnami-docker-keycloak/blob/master/15/debian-10/rootfs/opt/bitnami/scripts/libkeycloak.sh#L367

Reply:
perhaps for consistency with other env-vars within the container, I'll switch to `KEYCLOAK_DATABASE_VENDOR`? It feels weird seeing DB_VENDOR up against DATABASE_HOST etc inside the container. I'll add the logic as a name change will be a simple find and replace if we want to go back to DB_VENDOR

Reply:
Looking into it a bit more, it doesn't appear to be as simple as using that env-var.

I can see a bunch of .XML files in https://github.com/keycloak/keycloak-containers/tree/main/server/tools which i believe are set via the flags in the scripts from bitnami.

I added some bits for DB_VENDOR and KEYCLOAK_DB_VENDOR similar to the other vars but will pass KEYCLOAK_DB_VENDOR via the chart as its like the rest.

In order to make that var useful though additional things will need to be passed. Prehaps getting a base in to support postgres and something else by default would enable others to expand upon the externalDatabases as and when they want that?

Reply:
> I'll switch to KEYCLOAK_DATABASE_VENDOR? It feels weird seeing DB_VENDOR up against DATABASE_HOST etc inside the container

Yes please. The images that we expose to configure Keycloak base on users preferences should be prefixed with `KEYCLOAK_`.

> Looking into it a bit more, it doesn't appear to be as simple as using that env-var.

I expected that. A resilient logic to automate the configuration of certain apps can be challenging.

> In order to make that var useful though additional things will need to be passed

What thing are you referring to?

Reply:
It's around the configuration, just setting the DB_VENDOR alone won't be enough to connect to a specific DB for example. Looking at the links it appears we need to update the image to actually make use of the variable and configure Keycloak appropriately

Reply:
It works with the small patch (just a few lines) to the entry point shell script in the docker container. We use Bitnami Keycloak with Oracle, MariaDB, and H2 since the end of October 2022 in our Dev and Test environments without any issues...